lib/SIL/SILConstants.cpp

```diff
@@ -156,31 +162,50 @@ SymbolicValue::Kind SymbolicValue::getKind() const {
 /// Clone this SymbolicValue into the specified allocator and return the new
 /// version.  This only works for valid constants.
 SymbolicValue SymbolicValue::cloneInto(llvm::BumpPtrAllocator &allocator) const{
-  switch (getKind()) {
-  case Unknown:
-  case UninitMemory:
-  case Metatype:
-  case Function:
-  case Enum:
+  auto thisRK = representationKind;
+  // If this is an instruction kind, map onto the thing that will cause a copy.
+  if (thisRK == RK_Inst) {
+    auto *inst = value.inst;
+    if (isa<IntegerLiteralInst>(inst))
```

Comment:
would it make the code simpler if we always store int/float insts as APInt/APFloat, so that we need not deal with these inst types in the case of RK_Inst?

then it seems we only need to deal with StringLiteralInst -- by the same reasoning, if we convert that to RK_String, can we then remove the enum case of RK_Inst?